components/core/tests/test-ir_encoding_methods.cpp

```diff
@@ -332,6 +332,51 @@ TEMPLATE_TEST_CASE("decode_next_message_general", "[ffi][decode_next_message]",
             decode_next_message<TestType>(incomplete_message_buffer, message, timestamp));
 }
 
+// The test case only tests eight_byte_encoded_variable_t. Because the timestamp
+// size is fixed, it is easier to reverse engineer the last index of logtype in
+// the ir_buffer. The test makes modification to the logtype in the ir_buf
+// to trigger a mocked IRErrorCode_Decode_Error.
+TEST_CASE("message_decode_error", "[ffi][decode_next_message]")
+{
```

Comment:
```suggestion
// NOTE: This test only tests eight_byte_encoded_variable_t because we trigger
// IRErrorCode_Decode_Error by manually modifying the logtype within the IR, and
// this is easier for the eight_byte_encoded_variable_t case.
TEST_CASE("message_decode_error", "[ffi][decode_next_message]") {
```